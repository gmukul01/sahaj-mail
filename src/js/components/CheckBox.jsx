import React from 'react';

const CheckBox = ({ id, selected, onChange }) => (
	<>
		<input className="styled-checkbox" value={id} type="checkbox" id={`checkbox-${id}`} defaultChecked={selected} onChange={onChange} />
		<label htmlFor={`checkbox-${id}`} />
	</>
);

export default CheckBox;
