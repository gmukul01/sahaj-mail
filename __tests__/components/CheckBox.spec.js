import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CheckBox from 'components/CheckBox';

describe('CheckBox Component', () => {
	const onChangeHandler = jest.fn(),
		initialProps = {
			id: 1,
			selected: true,
			onChange: onChangeHandler
		};
	it('should match snapshot', () => {
		const checkBox = renderer.create(<CheckBox {...initialProps} />).toJSON();
		expect(checkBox).toMatchSnapshot();
	});

	it('should call onChange Handler on onChange event', () => {
		const checkBox = shallow(<CheckBox {...initialProps} />);
		checkBox.find('input').simulate('change', 'Dummy');
		expect(onChangeHandler.mock.calls[0][0]).toBe('Dummy');
	});
});
