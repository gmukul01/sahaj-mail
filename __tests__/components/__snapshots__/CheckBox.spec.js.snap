// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component should match snapshot 1`] = `
Array [
  <input
    className="styled-checkbox"
    defaultChecked={true}
    id="checkbox-1"
    onChange={[MockFunction]}
    type="checkbox"
    value={1}
  />,
  <label
    htmlFor="checkbox-1"
  />,
]
`;
